File: GJScrollViewUsage/GJScrollViewUsage/View/ParameterCell.m
//
//  ParameterCell.m
//  BuyCarDictionary
//
//  Created by zgjun on 14-12-29.
//  Copyright (c) 2014年 chexun. All rights reserved.
//

#import "ParameterCell.h"


@interface ParameterCell()
@property (nonatomic,weak) UILabel *showTitle;
@property (nonatomic,weak) UIView *rightLine;
@property (nonatomic,weak) UIView *bottomLine;

@property (nonatomic,strong) NSString *indicator;
@end

@implementation ParameterCell

- (instancetype)initWithFrame:(CGRect)frame indicator:(NSString *)indicator {
    if (self = [super initWithFrame:frame]) {
        self.indicator = indicator;
        //create child views
        UILabel *showTitle = [[UILabel alloc]init];
        showTitle.textAlignment = NSTextAlignmentCenter;
        showTitle.font = [UIFont systemFontOfSize:12];
        showTitle.numberOfLines = -1;
        self.showTitle = showTitle;
        [self addSubview:showTitle];
        //the right vertical line
        UIView *rightLine = [[UIView alloc]init];
        rightLine.backgroundColor = MainBackGroundColor;
        self.rightLine = rightLine;
        [self addSubview:rightLine];
        //the bottom line
        UIView *bottomLine = [[UIView alloc]init];
        bottomLine.backgroundColor = MainBackGroundColor;
        self.bottomLine = bottomLine;
        [self addSubview:bottomLine];
    }
    return self;
}

- (void)layoutSubviews {
    [super layoutSubviews];
    
    self.showTitle.frame = CGRectMake(0, 0, self.width, self.height);
    self.rightLine.frame = CGRectMake(self.width - 1, 0, 1, self.height);
    self.bottomLine.frame = CGRectMake(0, self.height - 1, self.width, 1);
    
}

- (void)setContentStr:(NSString *)contentStr {
    _contentStr = contentStr;
    self.showTitle.text = contentStr;
    
    CGFloat tipLabelX = 0;
    
    CGFloat tipLabelY = 0;
    
    CGSize maxSize = CGSizeMake(self.width, MAXFLOAT);
    
    CGSize tipLabelSize = [contentStr sizeWithFont:self.showTitle.font constrainedToSize:maxSize];
    
    self.showTitle.frame = (CGRect){{tipLabelX,tipLabelY},tipLabelSize};
}

- (void)setIsAll:(BOOL)isAll {
    _isAll = isAll;
    if (_isAll == YES) {
        self.showTitle.textColor = MainGoldenColor;
    } else {
        self.showTitle.textColor = [UIColor blackColor];
    }
    
    if ([self.indicator isEqualToString:@"title"]) {
        self.showTitle.textColor = [UIColor blackColor];
    }
    
}

@end

File: GJScrollViewUsage_Swift/GJScrollViewUsage_Swift/View/ParameterCell.swift
//
//  ParameterCell.swift
//  GJScrollViewUsage_Swift
//
//  Created by zgjun on 15/9/24.
//  Copyright (c) 2015年 chexun. All rights reserved.
//

import UIKit

class ParameterCell: UIView {
    
    let mainBackGroundColor : UIColor = UIColor(red: (232.0 / 255.0), green: (233.0 / 255.0), blue: (235.0 / 255.0), alpha: 1.0)
    let mainGoldenColor : UIColor = UIColor(red: (230.0 / 255.0), green: (185.0 / 255.0), blue: (44.0 / 255.0), alpha: 1.0)
    var contentStr : String! {
        set(newContentStr) {
            showTitle.text = newContentStr
            let tipLabelX : CGFloat = 0.0
            let tipLabelY : CGFloat = 0.0
            let maxSize : CGSize = CGSizeMake(self.bounds.size.width, CGFloat(MAXFLOAT))
            let contentStrTemp : NSString = newContentStr as NSString
            let tipLabelSize : CGSize = contentStrTemp.sizeWithAttributes([NSFontAttributeName: UIFont.systemFontOfSize(12.0)])
            showTitle.frame = CGRectMake(tipLabelX, tipLabelY, tipLabelSize.width, tipLabelSize.height)
        }
        get {
            return self.contentStr
        }
    }
    var isAll : Bool! {
        set(newIsAll) {
            if newIsAll == true {
                showTitle.textColor = mainGoldenColor
            } else {
                showTitle.textColor = UIColor.blackColor()
            }
            
            if indicator == "title" {
                showTitle.textColor = UIColor.blackColor()
            }
        }
        get {
            return self.isAll
        }
    }
    var indicator : String!
    
    //views
    var showTitle : UILabel!
    var rightLine : UIView!
    var bottomLine : UIView!
    
    var selfWidth : CGFloat!
    var selfHeight : CGFloat!
    
    
    init(Frame frame: CGRect,IndicatorNew indicatorNew : String) {
        super.init(frame: frame)
        indicator = indicatorNew
        showTitle = UILabel.new()
        showTitle.textAlignment = NSTextAlignment.Center
        showTitle.font = UIFont.systemFontOfSize(12)
        showTitle.numberOfLines = -1
        self.addSubview(showTitle)
        
        rightLine = UIView.new()
        rightLine.backgroundColor = mainBackGroundColor
        self.addSubview(rightLine)
        
        bottomLine = UIView.new()
        bottomLine.backgroundColor = mainBackGroundColor
        self.addSubview(bottomLine)
        
    }

    required init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func layoutSubviews() {
         super.layoutSubviews()
        
        showTitle.frame = CGRectMake(0, 0, self.bounds.size.width, self.bounds.size.height);
        rightLine.frame = CGRectMake(self.bounds.size.width - 1, 0, 1, self.bounds.size.height);
        bottomLine.frame = CGRectMake(0, self.bounds.size.height - 1, self.bounds.size.width, 1);
    }
    
}
